Criação das entidades

rafaelfelixmoraes opened the pull request:
Criação das entidades a serem utilizadas durante o projeto, juntamente com seus respectivos repositories, e ajustando disposição das pastas seguindo clean arch.

Geração das entidades pelo JPA:

```
Hibernate: create table cliente (id bigint not null, nome varchar(200), cpf varchar(255), email varchar(255), primary key (id))

Hibernate: create table pedido (senha_retirada integer, valor_total numeric(38,2), cliente_id bigint unique, id bigint not null, status_pedido varchar(255) check (status_pedido in ('RECEBIDO','EM_PREPARACAO','PRONTO','FINALIZADO','CANCELADO')), primary key (id))

Hibernate: create table pedido_produtos (pedido_id bigint not null, produtos_id bigint not null, primary key (pedido_id, produtos_id))

Hibernate: create table produto (valor_unitario numeric(38,2), categoria_id bigint not null, id bigint not null, descricao varchar(100), primary key (id))
```

Commit:
Criação das entidades e ajustando disposição das pastas seguindo clean arch

wen-dell commented on src/main/java/br/com/tech/challenge/domain/entidades/Categoria.java:
```diff
@@ -0,0 +1,38 @@
+@Builder(toBuilder = true)
+@AllArgsConstructor
+@NoArgsConstructor
+public class Categoria implements Serializable {
```
Qual o ganho de adicionar a interface Serializable? Entendo que antigamente ela era muito utilizada para permitir que objetos trafegassem de forma mais fácil na rede, mas hoje em dia me parece que não. Já vi algumas discussões a respeito e uns devs disseram que hoje em dia seria desnecessário.

rafaelfelixmoraes replied:
Sim, realmente quando nos referíamos a objetos que trafegam da memória para o banco de dados, não é mais uma prática utilizar. Porém vejo que ainda é uma boa prática para quando os dados estão trafegando da memória para o response. Removi a implementação da interface das entidades

wen-dell commented on src/main/java/br/com/tech/challenge/domain/entidades/Pedido.java:
```diff
@@ -0,0 +1,43 @@
+@Builder(toBuilder = true)
+@AllArgsConstructor
+@NoArgsConstructor
+public class Pedido implements Serializable {
```
Acho que é bom colocar um método equals e hashcode. Coloca para se um objeto tiver o mesmo id, eles são iguais. Vai ajudar porque está sendo utilizado o set, que não permite objetos iguais na estrutura.

rafaelfelixmoraes replied:
Feito. Inclui para todas as entidades.

Commit:
- Ajustando entidades no padrão JPA
- Ajuste scripts do migrations
- Ajustando properties

Commit:
Removendo imports não usados

wen-dell reviewed (commented):
Só dois ajustes

Commit:
- Incluindo annotation @EqualsAndHashCode e removendo a implementação do Serializable nas entidades
- Ajustando config do JPA, para não recriar ou deletar as tabelas na inicialização e finalização da aplicação

Commit:
Removendo imports não utilizados
